### PR TITLE
[ALLUXIO-3165] Generate ufsStatus from fingerprint instead of calling ufs again.

### PR DESCRIPTION
**Pull request by yuzhu**
https://alluxio.atlassian.net/browse/ALLUXIO-3165

Optimize UFS sync to extract UfsStatus information from the fingerprint, instead of retrieving the UfsStatus again

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang PTAL. The status we generate from the fingerprint is not going to be complete, because it is missing a few piece of information such as last modified time and content length that we normally get from `ufsStatus`.

**Comment by gpang** [comment]
@yuzhu I see. What should we do about the partial ufs status? I don't think that is ideal, so maybe we just get the metadata from the fingerprint, just like we can check the metadata of the fingerprint.

Also, can you add some unittests to validate the transformations?

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
@gpang added a test. PTAL.

**Comment by yuzhu** [comment]
jenkins, test this please

File: core/common/src/test/java/alluxio/underfs/FingerprintTest.java [hunk 1]
```diff
@@ -88,4 +89,31 @@ public void matchMetadataOrContent() {
...
+  public void createFingerprintFromUfsStatus() {
+    String name = CommonUtils.randomAlphaNumString(10);
+    String owner = CommonUtils.randomAlphaNumString(10);
+    String group = CommonUtils.randomAlphaNumString(10);
+    short mode = (short) mRandom.nextInt();
+    String ufsName = CommonUtils.randomAlphaNumString(10);
+
+    UfsDirectoryStatus dirStatus = new UfsDirectoryStatus(name, owner, group, mode);
+    Fingerprint fp = Fingerprint.create(ufsName, dirStatus);
+    assertEquals(fp.getTag(Fingerprint.Tag.OWNER), owner);
```
Comment (gpang) [inline core/common/src/test/java/alluxio/underfs/FingerprintTest.java]: NIT: I think for `assertEquals`, the first argument should be the expected value, and the second argument is the one being tested. (same for the rest in this test method)

Comment (yuzhu) [reply core/common/src/test/java/alluxio/underfs/FingerprintTest.java]: Done!

**Comment by yuzhu** [comment]
@gpang Thanks for the review! Comments addressed.

File: core/common/src/test/java/alluxio/underfs/FingerprintTest.java (hunk 1 above)
Comment (gpang) [reply core/common/src/test/java/alluxio/underfs/FingerprintTest.java]: Can you change these batch of `assertEquals` too?

**Pull request merged**